data/scenario-test.cfg

```diff
@@ -1261,32 +1261,33 @@ local result = wesnoth.synchronize_choice(function()
       } }
     }
 
-    local function preshow()
+    local function preshow(dialog)
         local function select()
-            local i = wesnoth.get_dialog_value "the_list"
+            local i = dialog.the_list.selected_index
             local ut = wesnoth.unit_types[types[i]].__cfg
             local profile = string.gsub(ut.profile, "([^/]+)$", "transparent/%1")
-            if wesnoth.get_dialog_value "flip" then profile = profile .. "~FL()" end
-            wesnoth.set_dialog_value(profile, "the_image")
+            if dialog.flip.selected_index == 1 then profile = profile .. "~FL()" end
+            dialog.the_image.label = profile
         end
-        wesnoth.set_dialog_callback(select, "the_list")
-        wesnoth.set_dialog_callback(select, "flip")
+        dialog.the_list.callback = select
+        dialog.the_list.callback = select
         for i,v in ipairs(types) do
             local ut = wesnoth.unit_types[v].__cfg
-            wesnoth.set_dialog_value(ut.name, "the_list", i, "the_label")
-            wesnoth.set_dialog_value(ut.image, "the_list", i, "the_icon")
+            dialog:find("the_list", i, "the_label").label = ut.name
+            dialog:find("the_list", i, "the_icon").label = ut.image
         end
-        wesnoth.set_dialog_value(2, "the_list")
+        dialog.the_list.selected_index = 2
         select()
+        -- TODO: How does this convert to the new API? It seems to be missing an argument...?
```

Comment:
The window itself is also a widget sao this probably just sets the canvas of  the window so probably just:
```
dialog:set_canvas(2, {
            T.rectangle { x = 20, y = 20, w = 20, h = 20, fill_color= "0,0,255,255" },
            T.rectangle { x = 30, y = 30, w = 20, h = 20, fill_color = "255,0,0,255" }
})
```